### ISSUE TITLE
support case changing of labels

### ISSUE THREAD

**Issue by Xiphe** [Jimdo/github-sync-labels-milestones#6]
**Scenario** I want to rename "mylabel" to "MYLABEL"

**Given** I have a label "mylabel"

**When** I add the following label declaration to my config file

``` json
{
  "name": "MYLABEL",
  "previousNames": [
    "mylabel"
  ]
}
```

**Then** I expect github-sync-labels-milestones to update the label

**But** It fails with

```
{"message":"Validation Failed","errors":[{"resource":"Label","code":"already_exists","field":"name"}],"documentation_url":"https://developer.github.com/v3/issues/labels/#create-a-label"}
```

---
### Background

Currently the renaming works as following:
- Create new label with new name
- Find all Issues labeled with old name
- Add new label to the Issues
- Delete label with old name
### Observation

GitHub is case-insensitive when checking if a label already exists and therefore does not allow creation of "MYLABEL" as long as "mylabel" still exists.

**Comment by Xiphe** [comment]
cc @klartexter @mfrawley

**Comment by Xiphe** [comment]
from what I see, there still is not convenient way to just rename the label via api 
(ref https://developer.github.com/v3/issues/labels/#update-a-label)

**Comment by Xiphe** [comment]
Workaround is execute once with:

``` json
{
  "name": "_MYLABEL",
  "previousNames": [
    "mylabel"
  ]
}
```

then switch to:

``` json
{
  "name": "MYLABEL",
  "previousNames": [
    "_MYLABEL"
  ]
}
```

but this is not future-prove, whenever you add a repo using `mylabel` later on, it wont get renamed.

Still works, as you can see in the label history above ;)

**Comment by Xiphe** [comment]
possible flow could be
- Find and cache all Issues labeled with old name
- Delete label with old name
- Create new label with new name
- Add new label to the Issues

but in case of a api failure that might just remove labels from issues whereas previously, the worst case would be to have a duplicated label on the issue...

might be worth it though.

**Comment by KlausTrainer** [comment]
Hi @Xiphe :)

We're using the [`github.issues.edit`](http://mikedeboer.github.io/node-github/#api-issues-edit) API call in [`renameLabels`](https://github.com/Jimdo/github-sync-labels-milestones/blob/master/lib/updateLabels.js#L80). I guess this won't work for renaming labels that aren't used in any issue. 

Have you tried using [`github.issues.updateLabel`](http://mikedeboer.github.io/node-github/#api-issues-updateLabel) instead? It looks like it's using the GitHub API endpoint that you were previously mentioning here (https://developer.github.com/v3/issues/labels/#update-a-label).

**Comment by Xiphe** [comment]
Hey @KlausTrainer 

yes, we're also using `updateLabel` ([here](https://github.com/Jimdo/github-sync-labels-milestones/blob/master/lib/updateLabels.js#L171)).

But as i've said previously, this can not be used to rename a label but only to change it's color. Since the api just accepts one single name attribute and we somehow need to specify which label we want to rename ;)

ref: https://github.com/mikedeboer/node-github/blob/master/lib/routes.json#L2085-L2095

From looking at the github API again, I now see that it should be possible when sending a `PATCH`
of `{ name: "bar", color: "ff00ff" }` to `https://api.github.com/repos/Jimdo/github-sync-labels-milestones/labels/foo`

will try that out later on

**Comment by KlausTrainer** [comment]
Yeah, I tried renaming a label by calling that endpoint using `curl`, and it worked :)

**Comment by KlausTrainer** [comment]
But thanks, now I understand the problem, I guess :)

**Comment by Xiphe** [comment]
great 🎉 !

I just monkey-patched node-github and it worked as well. Will propose a PR there and integrate later to fix this issue.

**Comment by KlausTrainer** [comment]
Yay :tada:

Let me know if you'd like me to have a look at the PR before proposing it!

**Comment by Xiphe** [comment]
ref: https://github.com/mikedeboer/node-github/issues/415

**Comment by Xiphe** [comment]
released as `v0.1.3`

**Comment by ghost** [comment]
Awesome! Nice contribution to a major release 🎯 
Thanks @Xiphe & @KlausTrainer for the fast action!
